src/service/resources-impl.js

```diff
@@ -866,12 +878,34 @@ export class Resources {
         const resource = request.resource;
         const box = resource.getLayoutBox();
         const iniBox = resource.getInitialLayoutBox();
-        const diff = request.newHeight - box.height;
+
+        let topMarginDiff = 0;
+        let bottomMarginDiff = 0;
+        let bottomDisplacedBoundary = box.bottom;
+        let newMargins = undefined;
+        if (request.marginChange) {
+          newMargins = request.marginChange.newMargins;
+          const margins = request.marginChange.currentMargins;
+          if (newMargins.top != undefined) {
+            topMarginDiff = newMargins.top - margins.top;
+          }
+          if (newMargins.bottom != undefined) {
+            bottomMarginDiff = newMargins.bottom - margins.bottom;
+          }
+          if (bottomMarginDiff) {
+            // The lowest boundary of the element that would appear to be
+            // resized as a result of this size change. If the bottom margin is
+            // being changed then it is the bottom edge of the margin box,
+            // otherwise it is the bottom edge of the layout box as set above.
+            bottomDisplacedBoundary = box.bottom + margins.bottom;
```

Comment:
No. It's supposed to be the current position of the bottom boundary that will be displaced. This is used in the check that the bottom of the element is above the viewport (or within the 10% offset allowed). As long as the bottom displaced boundary meets this criteria then it doesn't matter what happens in terms of height changes of top margin changes, since the scroll adjusting will act such that nothing appears to change in the viewport except possibly in that allowed 10% region.